grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -24,25 +26,59 @@ import org.pih.warehouse.product.Product
+import org.pih.warehouse.requisition.RequisitionType
 import org.pih.warehouse.shipping.Shipment
+import org.pih.warehouse.shipping.ShipmentStatusCode
 
 class StockMovementApiController {
 
-    StockMovementService stockMovementService
     def dataService
+    StockMovementService stockMovementService
 
     def list = {
-        int max = Math.min(params.max ? params.int('max') : 10, 1000)
-        int offset = params.offset ? params.int("offset") : 0
-        String orderBy = params.orderBy ?: "requisition.dateRequested"
-        StockMovementDirection stockMovementDirection = params.direction ? params.direction as StockMovementDirection : null
         Location destination = params.destination ? Location.get(params.destination) : null
+        Location origin = params.origin ? Location.get(params.origin) : null
+
+        StockMovementDirection direction = params.direction ? params.direction as StockMovementDirection : null
+        if (destination == origin || !params.direction) {
+            def message = "Direction parameter is required. Origin and destination cannot be the same."
+            response.status = 400
+            render([errorMessages: [message]] as JSON)
+            return
+        }
 
         StockMovement stockMovement = new StockMovement()
-        stockMovement.stockMovementDirection = stockMovementDirection
+        stockMovement.stockMovementDirection = direction
+        stockMovement.origin = origin
         stockMovement.destination = destination
+        stockMovement.requestedBy = params.requestedBy ? User.get(params.requestedBy) : null
+        stockMovement.createdBy = params.createdBy ? User.get(params.createdBy) : null
+        stockMovement.updatedBy = params.updatedBy ? User.get(params.updatedBy) : null
+        stockMovement.receiptStatusCodes = params.receiptStatusCode ? params?.list("receiptStatusCode") as ShipmentStatusCode[] : null
+        stockMovement.requisitionStatusCodes = params.requisitionStatusCode ? params?.list("requisitionStatusCode") as RequisitionStatus[] : null
+        stockMovement.requestType = params.requestType ? params.requestType as RequisitionType : null
+        stockMovement.sourceType = params.sourceType ? params.sourceType as RequisitionSourceType : null
+
+        if (params.q) {
+            stockMovement.identifier = "%" + params.q + "%"
+            stockMovement.name = "%" + params.q + "%"
+            stockMovement.description = "%" + params.q + "%"
+        }
+
+        if (params.format == 'csv') {
+            params.max = null
+            params.offset = null
+        }
+
+        def stockMovements = stockMovementService.getStockMovements(stockMovement, params)
 
-        def stockMovements = stockMovementService.getStockMovements(stockMovement, [max: max, offset: offset, orderBy: orderBy ])
+        if (params.format == 'csv' && stockMovements) {
+            def csv = getStockMovementsCsv(stockMovements)
```

Comment:
I don't think we should be doing CSV from the API. Why not do this in the stock movement controller?